diff --git a/pendulum_control/src/pendulum_demo.cpp b/pendulum_control/src/pendulum_demo.cpp
index ba0f72c..5c6f3b1 100644
--- a/pendulum_control/src/pendulum_demo.cpp
+++ b/pendulum_control/src/pendulum_demo.cpp
@@ -140,6 +140,9 @@ int main(int argc, char * argv[])
   auto controller_command_callback =
     [&pendulum_controller](pendulum_msgs::msg::JointCommand::ConstSharedPtr msg) -> void
     {
+      if (msg->clear_statistics) {
+        rttest_clear_statistics();
+      }
       pendulum_controller->on_pendulum_setpoint(msg);
     };
 
diff --git a/pendulum_control/src/pendulum_teleop.cpp b/pendulum_control/src/pendulum_teleop.cpp
index 5aef669..65ae9ea 100644
--- a/pendulum_control/src/pendulum_teleop.cpp
+++ b/pendulum_control/src/pendulum_teleop.cpp
@@ -17,6 +17,7 @@
 #include <fstream>
 #include <memory>
 #include <utility>
+#include <getopt.h>
 
 #include "rclcpp/rclcpp.hpp"
 
@@ -28,17 +29,42 @@ using namespace std::chrono_literals;
 
 // Non real-time safe node for publishing a user-specified pendulum setpoint exactly once
 
+void usage()
+{
+  printf("Usage: pendulum_teleop [options] [<angle>]\n\n"
+    "options:\n"
+    "  -h    Show this screen.\n"
+    "  -c    Clear all statistics.\n");
+}
+
 int main(int argc, char * argv[])
 {
   rclcpp::init(argc, argv);
 
   double command = M_PI / 2;
+  bool clear_statistics = false;
+
   if (argc < 2) {
     fprintf(
-      stderr,
-      "Command argument not specified. Setting command to 90 degrees (PI/2 radians).\n");
+        stderr,
+        "Command argument not specified. Setting command to 90 degrees (PI/2 radians).\n");
   } else {
-    command = atof(argv[1]);
+    int opt;
+
+    while ((opt = getopt(argc, argv, "ch")) != -1) {
+      switch (opt) {
+        case 'c':
+          clear_statistics = true;
+          break;
+        default:
+          usage();
+          return 1;
+      }
+    }
+
+    if (optind < argc) {
+      command = atof(argv[optind]);
+    }
   }
 
   auto teleop_node = rclcpp::Node::make_shared("pendulum_teleop");
@@ -50,6 +76,7 @@ int main(int argc, char * argv[])
 
   auto msg = std::make_unique<pendulum_msgs::msg::JointCommand>();
   msg->position = command;
+  msg->clear_statistics = clear_statistics;
 
   rclcpp::sleep_for(500ms);
   pub->publish(std::move(msg));
diff --git a/pendulum_msgs/msg/JointCommand.msg b/pendulum_msgs/msg/JointCommand.msg
index 96b6bbd..4f02d21 100644
--- a/pendulum_msgs/msg/JointCommand.msg
+++ b/pendulum_msgs/msg/JointCommand.msg
@@ -1 +1,2 @@
 float64 position
+bool clear_statistics false
